app/controllers/admin/videos_controller.rb

```diff
@@ -8,20 +8,36 @@ class Admin::VideosController < Admin::BaseController
   #   video.update(video_params)
   # end
 
-  def create
-    begin
-      tutorial  = Tutorial.find(params[:tutorial_id])
-      thumbnail = YouTube::Video.by_id(new_video_params[:video_id]).thumbnail
-      video     = tutorial.videos.new(new_video_params.merge(thumbnail: thumbnail))
-
-      video.save
+  # def create
+  #   begin
+  #     tutorial  = Tutorial.find(params[:tutorial_id])
+  #     thumbnail = YouTube::Video.by_id(new_video_params[:video_id]).thumbnail
+  #     video     = tutorial.videos.new(new_video_params.merge(thumbnail: thumbnail))
+  #     video.save
+  #
+  #     flash[:success] = "Successfully created video."
+  #   rescue # Sorry about this. We should get more specific instead of swallowing all errors.
+  #
+  #     flash[:error] = "Unable to create video."
+  #   end
+  #
+  #   redirect_to edit_admin_tutorial_path(id: tutorial.id)
+  # end
 
+  def create
```

Comment:
@nagerz Would love input here. Instead of rescuing the error, like when the Youtube Service can't find a video with an id that matches the video_id that the user entered, I added a conditional in the Video class below so that the thumbnail will just be nil. 

If the thumbnail is nil (because of a new validation I added for it in the Video model), then video.save will be false. 

I then just used video.errors.full_messages.to_sentence to add those error messages to the flash.

Reply:
Think it looks great. I took it and implemented it for tutorials too.